contracts/btc-staking/src/validation/mod.rs

```diff
@@ -49,6 +50,42 @@ fn verify_pop(
     Ok(())
 }
 
+#[cfg(feature = "full-validation")]
+fn get_pks(
+    staker_pk_hex: String,
+    fp_pk_hex_list: Vec<String>,
+    cov_pk_hex_list: Vec<String>,
```

Comment:
Use references to avoid early cloning.
```suggestion
    staker_pk_hex: &str,
    fp_pk_hex_list: &[String],
    cov_pk_hex_list: &[String],
```

or even `&[&str]` if possible / valid.